docs/usage/configuration/config_documentation.md

```diff
@@ -0,0 +1,3464 @@
+database defualts to SQLite, which is not recommended for production usage. You can read more on this subject 
+[here](../../setup/installation.md#using-postgresql).
+
+## Config Conventions
+
+Configuration options that take a time period can be set using a number
+followed by a letter. Letters have the following meanings:
+
+s = second
+
+m = minute
+
+h = hour
+
+d = day
+
+w = week
+
+y = year
+
+For example, setting `redaction_retention_period: 5m` would remove redacted
+messages from the database after 5 minutes, rather than 5 months.
+
+## Yaml 
+The configuration file is a [YAML](https://yaml.org/) file, which means that certain syntax rules
+apply if you want your config file to be read properly. A few helpful things to know:
+* `#` before any option in the config will comment out that setting and either a default (if available) will 
+   be applied or Synapse will ignore the setting. Thus, in example #1 below, the setting will be read and
+   applied, but in example #2 the setting will not be read and a default will be applied.  
+
+   Example #1:
+   ```
+   pid_file: DATADIR/homeserver.pid
+   ```
+   Example #2:
+   ```
+   #pid_file: DATADIR/homeserver.pid
+   ```
+* Indentation matters! The indentation before a setting
+  will determine whether a given setting is read as part of another
+  setting, or considered on its own. Thus, in example #1, the `enabled` setting
+  is read as a sub-option of the `presence` setting, and will be properly applied.
+  
+  However, the lack of indentation before the `enabled` setting in example #2 means
+  that when reading the config, Synapse will consider both `presence` and `enabled` as
+  different settings. In this case, `presence` has no value, and thus a default applied, and `enabled`
+  is an option that Synapse doesn't recognize and thus ignores.
+  
+  Example #1: 
+  ```
+  presence:
+    enabled: false
+  ```
+  Example #2:
+  ```
+  presence:
+  enabled: false
+  ```
+  In this manual, all top-level settings (ones with no indentation) are identified 
+  at the beginning of their section (i.e. "Config option: `example_setting`") and 
+  the sub-options, if any, are identified and listed in the body of the section. 
+  In addition, each setting has an example of its usage, with the proper indentation
+  shown. 
+
+## Index
+[Modules](#modules) 
+
+[Server](#server)
+
+[Homeserver Blocking](#homeserver-blocking)
+
+[TLS](#tls)
+
+[Federation](#federation)
+  
+[Caching](#caching)
+
+[Database](#database)
+
+[Logging](#logging)
+
+[Ratelimiting](#ratelimiting)
+
+[Media Store](#media-store)
+
+[Captcha](#captcha)
+
+[Turn](#turn)
+
+[Registration](#registration)
+
+[Metrics](#metrics)
+
+[API Configuration](#api-configuration)
+
+[Signing Keys](#signing-keys)
+
+[Single Sign On Integration](#single-sign-on-integration)
+
+[Push](#push)
+
+[Rooms](#rooms)
+
+[Opentracing](#opentracing)
+
+[Workers](#workers)
+
+[Background Updates](#background-updates)
```

Comment:
I don't think an index is necessary since the docs website already provides automatically.